sst/SimEngCoreWrapper.cc

```diff
@@ -0,0 +1,368 @@
+      // delimiting or escape behavior even with single quotes.
+      // e.g "arg1=1 arg2="James' Car" arg3=2" will be parsed as
+      // std::vector<std::string>{arg1=1, arg2=James' Car, arg3=2}
+      else if (currChar == '\"' && escapeDouble) {
+        escapeDouble = 0;
+      } else {
+        str += currChar;
+      }
+    } else {
+      if (currChar == ' ') {
+        if (str != "") {
+          args.push_back(str);
+          str = "";
+        }
+      }
+      // Check for escape character ("), this signals the algorithm to capture
+      // any char inside a set of ("") without producing any delimiting or
+      // escape behavior.
+      else if (currChar == '\"') {
+        escapeDouble = 1;
+        // Check for escape character ('), this signals the algorithm to capture
+        // any char inside a set of ('') without producing any delimiting or
+        // escape behavior.
+      } else if (currChar == '\'') {
+        escapeSingle = 1;
+      } else {
+        str += currChar;
+      }
+    }
+  }
+  if (escapeSingle || escapeDouble) {
+    std::string err;
+    output_.verbose(CALL_INFO, 1, 0, R"(
+           Parsing failed: Invalid format - Please make sure all
+           characters/strings are escaped properly within a set single or 
+           double quotes. To escape quotes use (\\\) instead of (\).\n
+           )");
+    std::cerr << "Error occured at index " << index
+              << " of the argument string - substring: "
+              << "[ " << str << " ]" << std::endl;
+    std::exit(EXIT_FAILURE);
+  }
+  args.push_back(str);
+  return args;
+}
+
+void SimEngCoreWrapper::fabricateSimEngCore() {
+  output_.verbose(CALL_INFO, 1, 0, "Setting up SimEng Core\n");
+  if (simengConfigPath_ != "") {
+#ifdef SIMENG_ENABLE_SST_TESTS
+    if (assembleWithSource_) {
+      output_.verbose(CALL_INFO, 1, 0,
+                      "Assembling source instructions using LLVM\n");
+      Assembler assemble = Assembler(source_);
+      coreInstance_ = std::make_unique<simeng::CoreInstance>(
+          assemble.getAssembledSource(), assemble.getAssembledSourceSize(),
+          simengConfigPath_);
+    } else {
+      coreInstance_ = std::make_unique<simeng::CoreInstance>(
+          simengConfigPath_, executablePath_, executableArgs_);
+    }
+#else
+    coreInstance_ = std::make_unique<simeng::CoreInstance>(
+        simengConfigPath_, executablePath_, executableArgs_);
+#endif
+  } else {
+#ifdef SIMENG_ENABLE_SST_TESTS
+    std::string a64fxConfigPath = std::string(SIMENG_BUILD_DIR) +
+                                  "/simeng-configs/sst-cores/a64fx-sst.yaml";
+    output_.verbose(
+        CALL_INFO, 1, 0,
+        "No config path provided so defaulting to a64fx-sst.yaml\n");
+    if (assembleWithSource_) {
+      output_.verbose(CALL_INFO, 1, 0,
+                      "Assembling source instructions using LLVM\n");
+      Assembler assemble = Assembler(source_);
```

Comment:
Is there any way that the source could be assembled inside the test suite and then passed into CoreWrapper? Could help make this function / class more test-suite-agnostic

Reply:
That's not possible because the test suite calls an sst with a config file.